engine/src/io/data_provider/UriDataProvider.cpp

```diff
@@ -186,17 +185,25 @@ data_handle uri_data_provider::get_next() {
 			return handle;
 		} else {
 			// this is a file we cannot parse apparently
-			return get_next();
+			this->current_file++;
```

Comment:
we shoudl be throwing some kind of warning here that we can somehow let the user know that something was aq little off. This might be one of the cases where a cout warning like the red ones that dask sometimes gives users would be the most appropriate

Reply:
The code only gets here if we are purposefully ignoring missing files. See my explanation above for why we are doing that. The code in this if/else branch is more about permitting continuing the traversal of the files and directories by the provider